### PR TITLE
Add the ability to white list plugins so we can turn off any potentia…

### PR DESCRIPTION
**Pull request by jaredadobe**
…l third party that may be killing our code

### CODE REVIEW

**Comment by jaredadobe** [comment]
This makes more sense with some Core PS code that i added.
This adds the ability for photoshop to tell generator to whitelist some plugins if we get into a state of external plugins causing exceptions.

**Comment by joelrbrandt** [comment]
@jaredadobe did you (or did someone) add code to automatically restart the node process if it crashes? I assume that's where the whitelisting would come in as being useful?

cc @iwehrman @mcilroyc

File: app.js [hunk 1]
```diff
@@ -237,11 +239,23 @@
...
+                if(shouldWhiteList) {
+                    whiteListedPlugins = argv.whiteListedPlugins.split(",").map(function(s) { return s.trim(); });
+                }
+
 
                 // Ensure all plugins have a valid semver, then put them in to a map
                 // keyed on plugin name
+
                 plugins.forEach(function (p) {
+                    if(shouldWhiteList && whiteListedPlugins.indexOf(p.metadata.name) == -1) {
```
Comment (joelrbrandt) [inline app.js]: need a `===` here

Comment (jaredadobe) [inline app.js]: Sure.

**Comment by jaredadobe** [comment]
@joelrbrandt yup. If you drop by the generator slack i outlined what happens.

**Comment by jaredadobe** [comment]
@joelrbrandt triple equals added!

**Comment by marekhrabe** [comment]
Hi, can anybody elaborate on this whitelist feature? In which situations it is going to be activated? How is the white list going to be managed? Is the fact that Generator has ignored and not loaded n-number of plugins going to be presented to the user? 

I'm little worried that this can silently deactivate any 3rd party plugins while on surface, for the user, it will look like Generator is fine and running. It might bring a lot of confusion.

**Comment by jaredadobe** [comment]
@marekhrabe this will alert the user that generator has crashed as it does now. While still running in the background with the whitelisted plugins. This is due to some internal features depending on generator now. 

If a third party plugin has crashed and the user needs it back then they can restart generator through the preferences pane (As they would have to now).

**Comment by marekhrabe** [comment]
OK, I understand. Can this whitelist mode be detected by JSX (or actually anything) in 3rd party Generator-dependent CEP panels so we can show message explaining user that Generator needs to be restarted in order to make that panel work? 

And my second question, maybe a little suggestion: would there be a more convenient way of restarting Generator other than *go to Preferences, Plug-ins, Disable Generator, OK, go to Preferences, Plug-ins, Enable Generator, OK*?

**Comment by jaredadobe** [comment]
@marekhrabe let me think about a way to do that. I can't promise anything for now, but will certainly try!

**Comment by joelrbrandt** [comment]
@jaredadobe Thanks for taking this issue on. The JS code looks fine to my eye -- I suspect the majority of the complexity is on the native side.

One minor nit: we should bump the minor version number, since the command line interface is part of the API.

I'll do that, and then I'll hit the merge button. Please stand by...

**Comment by joelrbrandt** [comment]
Errr, nevermind, it looks like we already bumped to 3.5.0 for the 16.1 release. MERGING!

**Pull request merged**